### PR TITLE
making listenip optional for the zabbix agent

### PR DESCRIPTION
**Pull request by tibers**
Addressing [Issue 473](https://github.com/voxpupuli/puppet-zabbix/issues/473) concerning the ListenIP being optional in the zabbix-agent config. Makes things much smoother for users installing zabbix into an image generated with packer, etc.

### CODE REVIEW

File: manifests/agent.pp [hunk 1]
```diff
@@ -303,7 +302,7 @@
       $listen_ip = $::ipaddress
     }
   } else {
-    $listen_ip = $::ipaddress
+    notify { 'listenip is undef and will not be written to the configuration file' : }
```
Comment (bastelfreak) [inline manifests/agent.pp]: this is not a good idea, because this will show up in each run as an applied resource. We should probably leave the variable undefined or empty string, and check for that in the template.

**Comment by bastelfreak** [comment]
Can you add a test that verifies that we don't write the empty argument into the config file?

**Comment by tibers** [comment]
OK that's actually where I get sketchy. I can do puppet code, but I don't know how to write tests. :\

**Comment by bastelfreak** [comment]
Hi @tibers, one example would be this: https://github.com/voxpupuli/puppet-zabbix/blob/master/spec/classes/agent_spec.rb#L204
Feel free to join our IRC channel #voxpupuli on freenode. We're always happy to assist you in writing tests!

**Comment by tibers** [comment]
We abandoned zabbix for sensu. Thanks for the tip though.
